tuf/api/metadata.py

```diff
@@ -808,17 +907,18 @@ def from_dict(cls, targets_dict: Mapping[str, Any]) -> "Targets":
         """Creates Targets object from its dict representation."""
         common_args = cls._common_fields_from_dict(targets_dict)
         targets = targets_dict.pop("targets")
-        delegations = targets_dict.pop("delegations")
+        delegations_dict = targets_dict.pop("delegations")
```

Comment:
if delegations is optional then this is clearly a bug.

This is properly fixed in #1367